libraries/eosiolib/contracts/eosio/key_value.hpp

```diff
@@ -462,6 +462,8 @@ namespace internal {
          return eosio::key_type{(char*)buffer, actual_value_size};
       }
 
+      bool valid() const { return itr_stat != status::iterator_erased; }
```

Comment:
Shouldn't this be `itr_stat == status::iterator_ok`?

Reply:
Should the iterator end() be true or false?

Reply:
> Shouldn't this be `itr_stat == status::iterator_ok`?

Updated.

Reply:
False.  You can't materialize the value from either erased or end.

So the use case for this is:
```c++
auto itr = find some value
if (itr)
   do something with the value
else
   create an entry for the key and value
```